### PR TITLE
Finally some tests from Team 2

### PR DESCRIPTION
**Pull request by pserrer**

### CODE REVIEW

File: tests/semantic/test1.invalid.mj [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/* -*- coding:utf-8; mode:java; -*- */
+
+class Test {
+
+	public int overflow() {
+		return 2147483648;
+	}
+
```
Comment (parttimenerd) [inline tests/semantic/test1.invalid.mj]: You should probably add a main method to clarify the intent of this test case.

File: tests/semantic/test1.valid.mj [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+	public X X;
+
+	public X X(X X) {
+		if (X == null) {
+			return new X();
+		}
+		return X(X(X(X)));
+	}
+
```
Comment (parttimenerd) [inline tests/semantic/test1.valid.mj]: This isn't valid MiniJava code, please add a main method.

File: tests/semantic/test2.invalid.mj [hunk 3]
```diff
@@ -0,0 +1,8 @@
+/* -*- coding:utf-8; mode:java; -*- */
+
+class Test {
+
+	public void f() {}
+	public void f() {}
+
```
Comment (parttimenerd) [inline tests/semantic/test2.invalid.mj]: Same goes here.

File: tests/semantic/test3.invalid.mj [hunk 4]
```diff
@@ -0,0 +1,9 @@
+/* -*- coding:utf-8; mode:java; -*- */
+
+class Test {
+
+	public void f() {
+		return f();  /* Valid C++, invalid (Mini)Java. */
+	}
+
```
Comment (parttimenerd) [inline tests/semantic/test3.invalid.mj]: And here.

File: tests/semantic/test3.valid.mj [hunk 5]
```diff
@@ -0,0 +1,10 @@
+/* -*- coding:utf-8; mode:java; -*- */
+
+class Test {
+
+	public int test() {
+		/* Not valid Java but valid MiniJava. */
+		return new int[3][new int[2][new int[1][0]]];
+	}
+
```
Comment (parttimenerd) [inline tests/semantic/test3.valid.mj]: And here.

**Comment by pserrer** [comment]
Added the missing main methods

**Pull request merged**